galaxy_ng/app/api/ui/views/sync.py

```diff
@@ -40,18 +39,21 @@ def post(self, request: Request, *args, **kwargs) -> Response:
                                     ' any registries associated with it.') % distro_path}
             )
 
-        for key, value in registry.get_connection_fields().items():
-            setattr(remote, key, value)
-        remote.save()
+        result = tasks.launch_container_remote_sync(remote, registry, distro.repository)
+        return OperationPostponedResponse(result, request)
+
+
+class ContainerSyncRegistryView(api_base.APIView):
+    permission_classes = [access_policy.ContainerRegistryRemoteAccessPolicy]
+    action = 'sync'
+
+    def post(self, request: Request, *args, **kwargs) -> Response:
+        registry = get_object_or_404(models.ContainerRegistryRemote, pk=kwargs['pk'])
 
         result = dispatch(
-            container_sync,
-            shared_resources=[remote],
-            exclusive_resources=[distro.repository],
+            tasks.sync_all_repos_in_registry,
```

Comment:
this would be a great usecase to use task groups, here is an example https://github.com/pulp/pulp_file/pull/536/files#diff-dc946cef2bc4fef289c4a2d72dde9e612e92efc300edb9273a4e01b914092401R212

Reply:
@ipanova thanks! I'll look into this.

Reply:
We're not totally sure if our task ui will work with task groups, but I'll give it a shot

Reply:
err-  i forgot this particular part returning task group response from viewset is unreleased but we do have task groups in pulp https://docs.pulpproject.org/pulpcore/restapi.html#tag/Task-Groups